### PR TITLE
Misc Refactoring/Improvements

### PR DESCRIPTION
**Pull request by lyndonhenry**
This pull request bundles a bunch of very small changes, each too minor to justify its own pull request but taken as a whole too major to simply bundle with another. 

They include:

- souffle-sh now includes a command to format with clang
- tabs have been replaced by four spaces
- fixes for certain misspelled words
- drop statements have been added to the ram program after stores, to free memory earlier in the case of lots of stores
- signed ints in PrecedenceGraph have been replaced by unsigned (the use of signed/unsiged here was previously inconsistent and unnecessary)
- removed now obsolete methods from RamStatements.h, as the IO system has handled their functionality for a while now
- a test has been added for issue #128, which can be uncommented in evaluation.at when that is solved (I've posted a comment on the relevant issue giving instructions to do this)
- added some nice utility methods in GraphUtils and PrecedenceGraph for design reasons

### CODE REVIEW

**Comment by mmcgr** [comment]
This is going to fail the style tests for several files. clang-format changed behaviour between 3.8 and 3.9 (I think you could call it a bugfix) It looks like you're using clang-format-4.0 or so, but the check uses 3.8.

I suggest you remove all the files where you are only changing style. They can be updated when the style checker is updated. That will be done when it looks like clang-format-3.9+ is more common.

**Comment by lyndonhenry** [comment]
Ahh, ok. Thankyou, that probably just saved me an evening of torture. I'll fix it up and push the changes.

File: src/PrecedenceGraph.h [hunk 1]
```diff
@@ -111,29 +116,50 @@ class SCCGraph : public AstAnalysis {
...
+            std::stack<const AstRelation*>& S, std::stack<const AstRelation*>& P, unsigned& numSCCs);
 
 public:
     static constexpr const char* name = "scc-graph";
 
     void run(const AstTranslationUnit& translationUnit) override;
 
-    int getSCCForRelation(const AstRelation* relation) {
-        return nodeToSCC[relation];
+    const unsigned getSCCForRelation(const AstRelation* relation) const {
```
Comment (mmcgr) [inline src/PrecedenceGraph.h]: Do these non-reference returns really need to return a const?

Comment (lyndonhenry) [inline src/PrecedenceGraph.h]: No there is no reason for them to do that and I shall change it back.

**Pull request merged**